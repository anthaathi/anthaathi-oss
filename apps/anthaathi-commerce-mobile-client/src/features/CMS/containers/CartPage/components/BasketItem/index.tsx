import {Image, Pressable, View} from 'react-native';
import React from 'react';
import {Divider, IconButton, Text, useTheme} from 'react-native-paper';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';
import {useIntl} from 'react-intl';
import {CartPageComponentType} from '../../../../types/common';
import {ProductProps} from '../../../ProductListPage/components/ProductList';

export interface ItemProps extends ProductProps {
  numberOfItems: number;
}

export interface BasketItemProps {
  title: string;
  items: ItemProps[];
  handlePressRemoveAllProduct?: () => void;
  handlePressViewProduct?: (item: ProductProps) => void;
  addProductPress?: (id: number) => void;
  removeProductPress?: (id: number, numberOfItems: number) => void;
}

const BasketItem = (props: BasketItemProps) => {
  const theme = useTheme();
  const intl = useIntl();
  const itemHeight = useResponsiveValue([120, 250, 290, 330]);
  const itemWidth = useResponsiveValue([120, 240, 280, 320]);

  return (
    <View
      style={{
        marginHorizontal: 5,
      }}
      testID="basketItem">
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}>
        <Text variant="titleLarge" style={{marginBottom: 9, fontSize: 20}}>
          {props.title}
        </Text>

        <Pressable
          onPress={props.handlePressRemoveAllProduct}
          testID="handlePressBasketItem">
          <Text
            variant="titleMedium"
            style={{
              marginBottom: 9,
              textDecorationLine: 'underline',
              fontSize: 14,
              color: theme.colors.greenTextColor,
            }}>
            {intl.formatMessage({defaultMessage: 'Remove All'})}
          </Text>
        </Pressable>
      </View>

      <View>
        {props.items.map(item => {
          return (
            <ItemRenderer
              key={item.key}
              item={item}
              itemHeight={itemHeight}
              itemWidth={itemWidth}
              handlePressViewProduct={
                props.handlePressViewProduct || (() => {})
              }
              addProductPress={props.addProductPress || (() => {})}
              removeProductPress={props.removeProductPress || (() => {})}
            />
          );
        })}
      </View>
    </View>
  );
};

const ItemRenderer = ({
  item,
  itemHeight,
  itemWidth,
  handlePressViewProduct,
  addProductPress,
  removeProductPress,
}: {
  item: ItemProps;
  itemHeight: number;
  itemWidth: number;
  handlePressViewProduct: (item: ProductProps) => void;
  addProductPress: (id: number) => void;
  removeProductPress: (id: number, numberOfItems: number) => void;
}) => {
  const theme = useTheme();
  const intl = useIntl();
  return (
    <View>
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
        }}>
        <View style={{flexDirection: 'row'}}>
          <Pressable onPress={() => handlePressViewProduct(item)}>
            <Image
              testID="basketProductImage"
              source={{uri: item.image}}
              style={{height: itemHeight, width: itemWidth}}
            />
          </Pressable>

          <View
            style={{
              marginHorizontal: 15,
              flexDirection: 'column',
              justifyContent: 'space-between',
              width: '100%',
            }}>
            <View style={{flexDirection: 'row', width: '45%'}}>
              <Text
                testID="productName"
                variant="titleLarge"
                style={{
                  flex: 1,
                  fontSize: 14,
                  color: theme.colors.titleTextColor,
                  fontWeight: '900',
                  flexWrap: 'wrap',
                }}>
                {item.name}
              </Text>
            </View>
            <ProductCountButton
              numberOfItems={item.numberOfItems}
              addProductPress={() => addProductPress(item.id)}
              removeProductPress={() =>
                removeProductPress(item.id, item.numberOfItems)
              }
            />

            <Text
              variant="titleLarge"
              style={{
                fontSize: 13,
                color: theme.colors.titleTextColor,
                fontWeight: '600',
              }}>
              {intl.formatNumber(item.price, {
                style: 'currency',
                currency: item.currency,
              }) +
                ' / ' +
                intl.formatMessage({defaultMessage: 'Piece'})}
            </Text>

            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
              }}>
              <Text
                variant="titleLarge"
                style={{
                  fontSize: 14,
                  color: theme.colors.titleTextColor,
                  fontWeight: '900',
                }}>
                {intl.formatMessage({defaultMessage: 'Total'}) + ' :'}
              </Text>

              <Text
                testID="productPrice"
                variant="titleLarge"
                style={{
                  fontSize: 14,
                  color: theme.colors.greenTextColor,
                  fontWeight: '700',
                  marginLeft: 5,
                }}>
                {intl.formatNumber(item.price * item.numberOfItems, {
                  style: 'currency',
                  currency: item.currency,
                })}
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Divider style={{marginVertical: 10}} />
    </View>
  );
};

const ProductCountButton = ({
  numberOfItems,
  addProductPress,
  removeProductPress,
}: {
  numberOfItems: number;
  addProductPress: () => void;
  removeProductPress: () => void;
}) => {
  const theme = useTheme();
  return (
    <View
      style={{
        width: 120,
        flexDirection: 'row',
        backgroundColor: '#F1F9F4',
        alignItems: 'center',
        justifyContent: 'center',
        height: 42,
      }}>
      <IconButton
        style={{
          borderRadius: 0,
          height: 42,
        }}
        icon={numberOfItems > 1 ? 'minus' : 'delete'}
        iconColor={theme.colors.greenTextColor}
        size={20}
        onPress={removeProductPress}
      />
      <Text
        style={{
          marginHorizontal: 5,
          fontSize: 14,
          color: theme.colors.titleTextColor,
          fontWeight: '700',
        }}>
        {numberOfItems}
      </Text>
      <IconButton
        style={{
          borderRadius: 0,
          height: 42,
        }}
        icon="plus"
        iconColor={theme.colors.greenTextColor}
        size={20}
        onPress={addProductPress}
      />
    </View>
  );
};

export default BasketItem;

export const BasketItemCMSInput = {
  _component: CartPageComponentType.BasketItem,
  component: BasketItem,
};
