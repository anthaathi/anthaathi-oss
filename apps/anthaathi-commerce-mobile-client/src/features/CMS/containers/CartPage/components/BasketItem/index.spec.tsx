import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import BasketItem from './index';

describe('BasketItem', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BasketItem
            title="Items"
            items={[
              {
                id: 1,
                name: 'Baby Yellow Pepper',
                image:
                  'https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1',
                key: '12',
                price: 12,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
              },
              {
                id: 2,
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('basketItem')).toBeTruthy();
  });

  it('should have BasketItem title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BasketItem
            title="Items"
            items={[
              {
                id: 1,
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );
    expect(temp.queryByText('Items')).toBeTruthy();
    expect(temp.queryByText('Remove All')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BasketItem
            title="Items"
            handlePressRemoveAllProduct={onpress}
            items={[
              {
                id: 1,
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
                notes: 'notes',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );
    fireEvent.press(temp.queryByTestId('handlePressBasketItem')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have BasketItem Product Image', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BasketItem
            title="Items"
            items={[
              {
                id: 1,
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
                notes: 'notes',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(
      temp.queryByTestId('basketProductImage')!.props.source,
    ).toMatchObject({
      uri: 'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
    });
  });

  it('should have BasketItem Product name, packing, price', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <BasketItem
            title="Items"
            items={[
              {
                id: 1,
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                numberOfItems: 2,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: '500 gms',
                notes: 'notes',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('productName')?.children).toContain(
      'Capsicum mixed',
    );
  });
});
