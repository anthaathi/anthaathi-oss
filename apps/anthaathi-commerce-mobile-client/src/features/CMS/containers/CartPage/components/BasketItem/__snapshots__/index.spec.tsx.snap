// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketItem should render the item 1`] = `
<View
  style={
    {
      "marginHorizontal": 5,
    }
  }
  testID="basketItem"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          {
            "fontSize": 20,
            "marginBottom": 9,
          },
        ]
      }
      variant="titleLarge"
    >
      Items
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="handlePressBasketItem"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "#008D3E",
              "fontSize": 14,
              "marginBottom": 9,
              "textDecorationLine": "underline",
            },
          ]
        }
        variant="titleMedium"
      >
        Remove All
      </Text>
    </View>
  </View>
  <View>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                {
                  "uri": "https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1",
                }
              }
              style={
                {
                  "height": 250,
                  "width": 240,
                }
              }
              testID="basketProductImage"
            />
          </View>
          <View
            style={
              {
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginHorizontal": 15,
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "width": "45%",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#364A15",
                      "flex": 1,
                      "flexWrap": "wrap",
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                testID="productName"
                variant="titleLarge"
              >
                Baby Yellow Pepper
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#F1F9F4",
                  "flexDirection": "row",
                  "height": 42,
                  "justifyContent": "center",
                  "width": 120,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                hitSlop={
                  {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "margin": 6,
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 15,
                        "height": 30,
                        "width": 30,
                      },
                      undefined,
                      {
                        "borderRadius": 0,
                        "height": 42,
                      },
                    ],
                  ]
                }
              >
                <View>
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 20,
                        },
                        [
                          {
                            "lineHeight": 20,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    Û∞ç¥
                  </Text>
                </View>
              </View>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginHorizontal": 5,
                    },
                  ]
                }
              >
                2
              </Text>
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                hitSlop={
                  {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "margin": 6,
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 15,
                        "height": 30,
                        "width": 30,
                      },
                      undefined,
                      {
                        "borderRadius": 0,
                        "height": 42,
                      },
                    ],
                  ]
                }
              >
                <View>
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 20,
                        },
                        [
                          {
                            "lineHeight": 20,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    Û∞êï
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#364A15",
                    "fontSize": 13,
                    "fontWeight": "600",
                  },
                ]
              }
              variant="titleLarge"
            >
              $12.00 / Piece
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                variant="titleLarge"
              >
                Total :
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#008D3E",
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginLeft": 5,
                    },
                  ]
                }
                testID="productPrice"
                variant="titleLarge"
              >
                $24.00
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            {
              "marginVertical": 10,
            },
          ]
        }
      />
    </View>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                {
                  "uri": "https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1",
                }
              }
              style={
                {
                  "height": 250,
                  "width": 240,
                }
              }
              testID="basketProductImage"
            />
          </View>
          <View
            style={
              {
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginHorizontal": 15,
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "width": "45%",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#364A15",
                      "flex": 1,
                      "flexWrap": "wrap",
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                testID="productName"
                variant="titleLarge"
              >
                Capsicum mixed
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#F1F9F4",
                  "flexDirection": "row",
                  "height": 42,
                  "justifyContent": "center",
                  "width": 120,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                hitSlop={
                  {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "margin": 6,
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 15,
                        "height": 30,
                        "width": 30,
                      },
                      undefined,
                      {
                        "borderRadius": 0,
                        "height": 42,
                      },
                    ],
                  ]
                }
              >
                <View>
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 20,
                        },
                        [
                          {
                            "lineHeight": 20,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    Û∞ç¥
                  </Text>
                </View>
              </View>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginHorizontal": 5,
                    },
                  ]
                }
              >
                2
              </Text>
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                hitSlop={
                  {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "margin": 6,
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 15,
                        "height": 30,
                        "width": 30,
                      },
                      undefined,
                      {
                        "borderRadius": 0,
                        "height": 42,
                      },
                    ],
                  ]
                }
              >
                <View>
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 20,
                        },
                        [
                          {
                            "lineHeight": 20,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    Û∞êï
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "#364A15",
                    "fontSize": 13,
                    "fontWeight": "600",
                  },
                ]
              }
              variant="titleLarge"
            >
              $23.00 / Piece
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#364A15",
                      "fontSize": 14,
                      "fontWeight": "900",
                    },
                  ]
                }
                variant="titleLarge"
              >
                Total :
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#008D3E",
                      "fontSize": 14,
                      "fontWeight": "700",
                      "marginLeft": 5,
                    },
                  ]
                }
                testID="productPrice"
                variant="titleLarge"
              >
                $46.00
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            {
              "marginVertical": 10,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
